src/heart.c

```diff
@@ -462,6 +482,9 @@ static int message_loop()
             timeout.tv_sec = min(timeout.tv_sec, init_handshake_end_time - now);
 
         if ((i = select(max_fd + 1, &read_fds, NULLFDS, NULLFDS, &timeout)) < 0) {
+            if (errno == EINTR)
+                continue;
```

Comment:
I don't understand what this is for?

Reply:
select returns whenever the process gets a signal. Heart didn't expect to get any signals before so they left this out.